pilot/pkg/bootstrap/configcontroller.go

```diff
@@ -142,7 +142,14 @@ func (s *Server) initConfigController(args *PilotArgs) error {
 	s.environment.IstioConfigStore = model.MakeIstioStore(s.configController)
 
 	if features.EnableIOR {
-		ior.Register(s.kubeClient, s.configController, args.Namespace, s.mrc)
+		s.addStartFunc(func(stop <-chan struct{}) error {
+			go leaderelection.
+				NewLeaderElection(args.Namespace, args.PodName, leaderelection.StatusController, s.kubeClient).
```

Comment:
all other components that use this seem to use a unique `electionID`, should we do the same?

Reply:
Not unique. `leaderelection.StatusController` is used more than one time for example. It's ok to use it as well.

Reply:
really? I could only find it being used once, in `pilot/pkg/bootstrap/configcontroller.go:399`

Reply:
```
$ rg NewLeaderElection
pilot/pkg/bootstrap/validation.go
83:				NewLeaderElection(args.Namespace, args.PodName, leaderelection.ValidationController, s.kubeClient).

pilot/pkg/bootstrap/configcontroller.go
98:				NewLeaderElection(args.Namespace, args.PodName, leaderelection.IngressController, s.kubeClient.Kube()).
369:			NewLeaderElection(args.Namespace, args.PodName, leaderelection.AnalyzeController, s.kubeClient).
399:				NewLeaderElection(args.Namespace, args.PodName, leaderelection.StatusController, s.kubeClient).

pilot/pkg/bootstrap/server.go
1075:				NewLeaderElection(args.Namespace, args.PodName, leaderelection.NamespaceController, s.kubeClient.Kube()).

```

Reply:
Done. Changed the id and tested. Worked fine.